integration/js/utils/SdkBaseTest.js

```diff
@@ -109,11 +120,33 @@ class SdkBaseTest extends KiteBaseTest {
     const numberOfSeleniumSessions = this.numberOfSessions(this.capabilities.browserName);
 
     console.log(`Provisioning ${numberOfSeleniumSessions} sessions on ${process.env.SELENIUM_GRID_PROVIDER}`);
-    this.seleniumSessions = [];
     for (let i = 0; i < numberOfSeleniumSessions; i++) {
-      const session = await this.createSeleniumSession(this.capabilities);
-      await session.init();
-      this.seleniumSessions.push(session)
+      try {
+        const session = await this.createSeleniumSession(this.capabilities);
+        await session.init();
+        this.seleniumSessions.push(session)
+      } catch (e) {
+        console.log('Failed to initialize');
+        console.log(e);
+        await emitMetric(this.testName, this.capabilities, 'E2E', 0);
+        await this.updateSeleniumTestResult();
+        await this.quitSeleniumSessions();
+        return;
+      }
+    }
+    console.log('I am sending that I am ready');
+    this.io.emit('session_ready');
+
+    //Wait for other to be ready
+    if (this.numberOfParticipant > 1) {
+      await this.waitForTestReady();
```

Comment:
What would happen in this case:
Thread 1 acquired a session on saucelabs and failed because the other thread was not ready. At this point the `numberAttendeeSessionReady` in server.js is 1.

Now the Thread 2 is able to acquire a session on saucelabs so the `numberAttendeeSessionReady` is now 2. The server.js will then send the message to Thread 1 (already ended) and Thread 2 (alive) that the tests are ready so Thread 2 will continue with the test.
We need a need a way to prevent this from happening.